src/ObjCRuntime/Class.cs

```diff
@@ -646,25 +646,56 @@ internal static bool IsCustomType (Type type)
 		}
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
-		internal static extern IntPtr objc_allocateClassPair (IntPtr superclass, string name, IntPtr extraBytes);
+		static extern IntPtr objc_allocateClassPair (IntPtr superclass, IntPtr name, IntPtr extraBytes);
+
+		internal static IntPtr objc_allocateClassPair (IntPtr superclass, string name, IntPtr extraBytes)
+		{
+			using var namePtr = new TransientString (name);
+			return objc_allocateClassPair (superclass, namePtr, extraBytes);
+		}
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
-		internal static extern IntPtr objc_getClass (string name);
+		static extern IntPtr objc_getClass (IntPtr name);
+
+		internal static IntPtr objc_getClass (string name)
+		{
+			using var namePtr = new TransientString (name);
+			return objc_getClass (namePtr);
+		}
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
 		internal static extern void objc_registerClassPair (IntPtr cls);
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
 		[return: MarshalAs (UnmanagedType.U1)]
-		internal static extern bool class_addIvar (IntPtr cls, string name, IntPtr size, byte alignment, string types);
+		static extern bool class_addIvar (IntPtr cls, IntPtr name, IntPtr size, byte alignment, IntPtr types);
+
+		internal static bool class_addIvar (IntPtr cls, string name, IntPtr size, byte alignment, string types)
+		{
+			using var namePtr = new TransientString (name);
+			using var typesPtr = new TransientString (types);
+			return class_addIvar (cls, namePtr, size, alignment, typesPtr);
+		}
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
 		[return: MarshalAs (UnmanagedType.U1)]
-		internal static extern bool class_addMethod (IntPtr cls, IntPtr name, IntPtr imp, string types);
+		static extern bool class_addMethod (IntPtr cls, IntPtr name, IntPtr imp, IntPtr types);
+
+		internal static bool class_addMethod (IntPtr cls, IntPtr name, IntPtr imp, string types)
+		{
+			using var typesPtr = new TransientString (types);
+			return class_addMethod (cls, name, imp, typesPtr);
+		}
 
 		[DllImport (Messaging.LIBOBJC_DYLIB)]
 		[return: MarshalAs (UnmanagedType.U1)]
-		internal extern static bool class_addMethod (IntPtr cls, IntPtr name, Delegate imp, string types);
+		internal extern static bool class_addMethod (IntPtr cls, IntPtr name, Delegate imp, IntPtr types);
+
+		internal  static bool class_addMethod (IntPtr cls, IntPtr name, Delegate imp, string types)
```

Comment:
Utterly minor:

```suggestion
		internal static bool class_addMethod (IntPtr cls, IntPtr name, Delegate imp, string types)
```

Reply:
Looks like the autoformatter took care of this! https://github.com/xamarin/xamarin-macios/commit/58d81634b807e202431399fcb9e2bb7e0ce2fc4e

Reply:
Reformat bots roll out!